stuff2sort/System.py

```diff
@@ -0,0 +1,1150 @@
+        self.qmatomslist = self.read_atoms_list(self.input_dict['qmatomslist'])
+        self.activeatomslist = self.read_atoms_list(self.input_dict['activeatomslist'])
+        if input_dict['useinnerouter']:
+            self.inneratomslist = self.read_atoms_list(self.input_dict['inneratomslist'])
+            self.outeratomslist = self.read_atoms_list(self.input_dict['outeratomslist'])
+        #   If we're not using inner/outer we're keeping the lists empty (currently necessary for some functions but that might change later)
+        else:
+            self.inneratomslist = []
+            self.outeratomslist = []
+
+        #   Read Connectivity
+        self.connectivity_list = self.read_connectity_from_topology(self.input_dict['topologyfile'])
+
+        #   Read Initial Geometry
+        #   XX AJ I would prefer only one function here independent of the file type and only make that distinction within the function. I'll get back to that later when I'm writing GeneratorGeometries.py
+        if self.input_dict['coordinatefile'][-4:] == ".gro":
+            #logger(logfile, "Reading geometry (.gro)...\n")
+            # XX AJ if we rewrite gro files to g96 files, the following function can be deleted and we can read the g96 file with geometry.readg96 afterwards
+            self.initial_geometry = geometry.readgeo(self.input_dict['coordinatefile'])
+            # Writing high-precision coordinate file
+            # logger(logfile, "Writing high-precision coordinate file...")
+            self.write_highprec(self.input_dict['coordinatefile']) # XX temp removed logfile until logfile decision of Florian AJ
+            self.input_dict['coordinatefile'] = self.input_dict['jobname'] + ".g96"
+            # logger(logfile, "Done.\n")
+        elif self.input_dict['coordinatefile'][-4:] == ".g96":
+            #logger(logfile, "Reading geometry (.g96)...\n")
+            self.initial_geometry = geometry.readg96(self.input_dict['coordinatefile'])
+
+        self.number_of_atoms = int(len(self.initial_geometry)/3)
+
+        #   Create xyzq (Coordinates And Charges) For The Whole System
+        #   XX AJ same here, I'll make one function of it
+        #   XX AJ technically, I would prefer this xyzq function not in this class, but it's used in the get_linkatoms_ang, so I'll keep it here
+        if input_dict['useinnerouter']:
+            self.xyzq = geometry.make_xyzq_io(self.initial_geometry, self.charge_vector, self.outeratomslist)
+        else:
+            self.xyzq = geometry.make_xyzq(self.initial_geometry, self.charge_vector)
+
+        #   Read linkatoms and next order atoms
+        self.m1list = self.identify_m1()
+        self.m2list = self.identify_m2()
+
+        self.linkatoms = get_linkatoms_ang(self.xyzq, self.qmatomslist, self.m1list, self.connectivity_list, [])
+        self.linkcorrlist, self.q1list, self.q2list, self.q3list, self.m3list = self.get_linkcorrlist()
```

Comment:
self.list_correlation_atoms_link (does corr stand for correlation? correction?)
self.list_coordinates_atoms_q1
self.list_coordinates_atoms_q2
self.list_coordinates_atoms_q3
self.list_coordinates_atoms_m3

Reply:
To be honest, I don't know what corr means, but I will check with Jan to save time.

Reply:
I also saw that the other lists don't contain coordinates but only atom indices. I changed the names to self.list_atoms_*

Reply:
Changed linkcorrlist to list_link_correction, it contains atom pairs for which Coulomb potential needs to be excluded.